### ISSUE TITLE
Cannot Use in Other Bazel Project

### ISSUE THREAD

**Issue by rauljordan** [prysmaticlabs/go-bls#19]
Hi all,

Although this project builds well with Bazel, we are having difficulty using it in another Bazel project (https://github.com/prysmaticlabs/prysm) as a dependency. 

The failure is as follows:
```
ERROR: /home/nishant/.cache/bazel/_bazel_nishant/2d55d16d3d0901116c0ae97899c3d124/external/com_github_prysmaticlabs_go_bls/BUILD.bazel:3:1: C++ compilation of rule '@com_github_prysmaticlabs_go_bls//:go_default_library%linux_amd64%cgo_c_lib' failed (Exit 1) gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -MD -MF ... (remaining 66 argument(s) skipped)

crypto.go:5:10: fatal error: external/herumi_mcl/include/mcl/bls.h: No such file or directory
compilation terminated.
```

Even after adding the transitive deps the Prysm WORKSPACE:
```
go_repository(
    name = "com_github_prysmaticlabs_go_bls",
    commit = "a37e65522727d6a8518151ce9008aa155632c13d",
    importpath = "github.com/prysmaticlabs/go-bls",
)

load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")

git_repository(
    name = "herumi_mcl",
    commit = "79b3a33e21072712f00985ed2adf34b3bcf0d74e",
    remote = "https://github.com/prysmaticlabs/mcl",
)

git_repository(
    name = "bazelify_gmp",
    commit = "bb4881b35e6864c90493980d035e1d984cafd093",
    remote = "https://github.com/robin-thomas/bazelify-gmp",
)

git_repository(
    name = "boringssl",
    commit = "fafc4482e85c09e7af5f71b2eb287b73ccd1020a",
    remote = "https://github.com/google/boringssl",
)

git_repository(
    name = "io_bazel_rules_m4",
    commit = "2bf69df77dfb6b3ba6b7fc95c304b0dc279375bc",
    remote = "https://github.com/jmillikin/rules_m4",
)

load("@io_bazel_rules_m4//:m4.bzl", "m4_register_toolchains")

m4_register_toolchains()
```

Any help on this would be much appreciated.

**Comment by rauljordan** [comment]
@robin-thomas do you have any suggestions? We think this might have to do with some issues of relative paths? Unsure

**Comment by prestonvanloon** [comment]
Relevant PR: https://github.com/prysmaticlabs/prysm/pull/1367

**Comment by robin-thomas** [comment]
@rauljordan Seems like a path issue.

In your prysm project workspace, there should a directory with name `bazel-<project name>`. Find out the path of `bls.h` inside that directory. It may not be matching `external/herumi_mcl/include/mcl/bls.h`.

**Comment by prestonvanloon** [comment]
From the sandbox: `/home/preston/.cache/bazel/_bazel_preston/2bf51e8690558a83a0a0514ab422fd4b/sandbox/linux-sandbox/1/execroot/__main__/external/herumi_mcl/include/mcl/bls.h`

From bazel-prysm: `bazel-prysm/external/herumi_mcl/include/mcl/bls.h`


`@com_github_prysmaticlabs_go_bls//:go-lib` builds 
`@com_github_prysmaticlabs_go_bls//:go_default_library` does not build

**Comment by rauljordan** [comment]
Looks like the path does match, which is weird that we still get the failure

**Comment by robin-thomas** [comment]
@prestonvanloon Can you turn on the logging in sandbox and post the whole console log when you try to build `go_default_library`?

**Comment by prestonvanloon** [comment]
[output.log](https://github.com/prysmaticlabs/go-bls/files/2790106/out.log)

@robin-thomas

**Comment by robin-thomas** [comment]
@prestonvanloon Do the following changes and try building again.

In `WORKSPACE` file in prysm, change `go-bls` to local repository.
```
local_repository(
    name = "go-bls",
    path = <path to the go-bls directory> 
)
```

Modify the `crypto.go` file inside `go-bls` directory from `#include "external/herumi_mcl/include/mcl/bls.h"` to `#include "include/mcl/bls.h"`

**Comment by prestonvanloon** [comment]
@robin-thomas That fixed it! See PR #20. Thanks!!

**Comment by robin-thomas** [comment]
@prestonvanloon Great!

But for the future, please find a better way because Bazel recommends using paths like `external/herumi_mcl/include/mcl/bls.h` (probably for namespace issue).

**Comment by rauljordan** [comment]
Hey @robin-thomas, just a few issues: so the hotfix from before allowed the system to work and we can build it fine in other bazel projects. However, IDEs such as Goland will show anything related to go-bls as red (unrecognized import) when using it in files. However, if I go into `bazel-prysm` and modify the cgo header include to match `external/herumi_mcl/include/bls.h`, Goland no longer complains and it has full, nice support for the go-bls project. However, bazel build //... for the project will fail saying this:
```
crypto.go:5:49: fatal error: external/herumi_mcl/include/mcl/bls.h: No such file or directory
```
So to recap, changing to external in the cgo include causes IDEs to be happy but the actual build to fail. Keeping the include to `include/mcl/bls.h` causes the build to be happy but IDEs to go crazy with red imports. Any suggestions? This is really hampering our dev workflow if we were to include go-bls.

**Comment by rauljordan** [comment]
Tagging @prestonvanloon @nisdas

**Comment by rauljordan** [comment]
Unrelated to this - this is a bug in rules_go with bazel support in Goland
